### PR TITLE
Fix a few invalid noqa comments

### PR DESCRIPTION
**Pull request by Avasam**
Extracted from #11496

### CODE REVIEW

File: stubs/WebOb/webob/exc.pyi [hunk 1]
```diff
@@ -41,7 +41,7 @@ class WSGIHTTPException(Response, HTTPException):
     def generate_response(self, environ: WSGIEnvironment, start_response: StartResponse) -> Iterable[bytes]: ...
     @property
     def wsgi_response(self) -> Self: ...  # type:ignore[override]
```
Comment (JelleZijlstra) [inline stubs/WebOb/webob/exc.pyi:43]:
```suggestion
    def wsgi_response(self) -> Self: ...  # type: ignore[override]
```

File: stubs/WebOb/webob/exc.pyi [hunk 2]
```diff
@@ -41,7 +41,7 @@ class WSGIHTTPException(Response, HTTPException):
...
-    def __str__(self) -> str: ...  # type:ignore[override]  # noqaY029
+    def __str__(self) -> str: ...  # type:ignore[override]  # noqa: Y029
```
Comment (JelleZijlstra) [inline stubs/WebOb/webob/exc.pyi:44]:
```suggestion
    def __str__(self) -> str: ...  # type: ignore[override]  # noqa: Y029
```

**Review by JelleZijlstra — COMMENTED** [review]
Thanks, suggested two other changes while we're here

File: stubs/WebOb/webob/exc.pyi (hunk 1 above)
Comment (Avasam) [reply stubs/WebOb/webob/exc.pyi:43]: I've seen (and done) that often. Idk if it's just outside a formatter's purview (Ruff formatter/black). Maybe Ruff linting and/or flake8-pyi could check for it.

Comment (JelleZijlstra) [reply stubs/WebOb/webob/exc.pyi:43]: Yes, this could be a good lint rule.

Comment (Avasam) [reply stubs/WebOb/webob/exc.pyi:43]: There's 40 instances of `type:ignore` in the code. I could fix them all at once here if you'd like.

Comment (JelleZijlstra) [reply stubs/WebOb/webob/exc.pyi:43]: Sure!

**Pull request merged**